File: tmp_clean_react_imports.py
from pathlib import Path

root = Path(r'c:\Users\user\KTPGV\src')
for path in root.rglob('*'):
    if path.suffix.lower() not in {'.ts', '.tsx', '.js', '.jsx'}:
        continue
    text = path.read_text(encoding='utf-8')
    new = text.replace("import React from 'react';\n", "").replace('import React from "react";\n', '').replace('import React, {', 'import {')
    if new != text:
        path.write_text(new, encoding='utf-8')
        print(path)

File: tmp-clean-react-imports.js
const fs = require('fs');
const path = require('path');

const root = path.join(process.cwd(), 'src');

function walk(dir) {
  for (const entry of fs.readdirSync(dir, { withFileTypes: true })) {
    const full = path.join(dir, entry.name);
    if (entry.isDirectory()) {
      walk(full);
      continue;
    }
    if (!['.ts', '.tsx', '.js', '.jsx'].includes(path.extname(entry.name))) continue;

    const text = fs.readFileSync(full, 'utf8');
    let next = text
      .replace(/import\s+React\s+from\s+['"]react['"];?\n/g, '')
      .replace(/import\s+React\s*,\s*\{/g, 'import {');

    if (next !== text) {
      fs.writeFileSync(full, next, 'utf8');
      console.log('updated', path.relative(process.cwd(), full));
    }
  }
}

walk(root);
